modules/katello_gpg_key.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# (c) 2018, Baptiste Agasse <baptiste.agagsse@gmail.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: katello_gpg_key
+short_description: Create and Manage Katello GPG keys
+description:
+  - Create and Manage Katello GPG keys
+author: "Baptiste Agasse (@bagasse)"
+requirements:
+  - "nailgun >= 0.28.0"
+  - "python >= 2.6"
+  - "ansible >= 2.3"
+options:
+  server_url:
+    description:
+      - URL of Foreman server
+    required: true
+  username:
+    description:
+     - Username on Foreman server
+    required: true
+  password:
+    description:
+     - Password for user accessing Foreman server
+    required: true
+  verify_ssl:
+    description:
+      - Verify SSL of the Foreman server
+    default: true
+    type: bool
+  name:
+    description:
+      - Name of the GPG key
+    required: true
+  organization:
+    description:
+      - Organization name that the GPG key is in
+    required: true
+  content:
+    description:
+    - Content of the GPG key
+    required: true
+  state:
+    description:
+      - State of the GPG Key.
+    default: present
+    choices:
+      - present
+      - absent
+'''
+
+EXAMPLES = '''
+- name: "Create katello client GPG key"
+  katello_gpg_key:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    name: "RPM-GPG-KEY-my-repo"
+    organization: "Default Organization"
+    content: "-----BEGIN PGP PUBLIC KEY BLOCK-----
```

Comment:
I preferred, if we did not include an actual gpg-key in this repo.
And for the example, it might be as instructive and much more readable if we did something like:
`content: "{{ lookup('file', 'key.asc') }}"`
I think this applies to the tests, too. Katello does not rigogrously validate, that you insert a valid key, iirc.